### PR TITLE
Fixed: Tx panics should not stop miner (#520)

### PR DESCRIPTION
**Pull request by VenkatTeja**
## Motivation
The is pull request solves the issue raised here: https://github.com/OLSF/libra/issues/520

## Test Plan
A test has been written at https://github.com/VenkatTeja/libra/blob/256ee6ca7aaa0590607790f7f4bfe5f54051d036/ol/miner/tests/integration-submit-tx.rs

The test tries to block the client port using iptables and after sometime checks if miner was able to mine more blocks than before blocking port. Then the port is unblocked and the tests waits for miner to submit the backlog. The test checks again after sometime to check if client is in sync with miner.

### CODE REVIEW

File: ol/miner/src/block.rs [hunk 1]
```diff
@@ -104,32 +105,87 @@ pub fn mine_and_submit(
...
-
-            if let Some(ref _node) = config.profile.default_node {
-                match commit_proof_tx(&tx_params, block.preimage, block.proof, is_operator) {
-                    Ok(tx_view) => match eval_tx_status(tx_view) {
-                        Ok(_) => status_ok!("Success:", "Proof committed to chain"),
-                        Err(_) => status_err!("Miner transaction rejected"),
-                    },
-                    Err(err) => status_err!("Miner transaction rejected: {}", err),
+            
+            // submits backlog to client
```
Comment (lpgeiger) [inline ol/miner/src/block.rs:137]: We already have a method called backlog::process_backlog(), is there a reason we are not using that here?

File: ol/miner/src/commit_proof.rs [hunk 2]
```diff
@@ -18,10 +18,10 @@ pub fn commit_proof_tx(
     // Create a client object
     let mut client = LibraClient::new(tx_params.url.clone(), tx_params.waypoint).unwrap();
 
-    let chain_id = ChainId::new(client.get_metadata().unwrap().chain_id);
+    let chain_id = ChainId::new(client.get_metadata()?.chain_id);
```
Comment (lpgeiger) [inline ol/miner/src/commit_proof.rs:21]: I think chain_id is already in TxParams, can you check?

File: ol/miner/src/block.rs [hunk 3]
```diff
@@ -104,32 +105,87 @@ pub fn mine_and_submit(
...
         loop {
+            status_info!(
+                "Latest client block:",
+                format!("{}", remote_height)
+            );
+
             status_info!(format!("Block {}", mining_height), "Mining VDF Proof");
 
-            let block = mine_once(&config)?;
+            let block_result = mine_once(&config);
```
Comment (lpgeiger) [inline ol/miner/src/block.rs:122]: If you just call backlog::process_backlog() below, you can simplify this match and just panic.

```
match mine_once(&config) {

                Ok(blk) => block = blk,
                Err(err) => {
                    status_warn!("Mining block {} failed: {}", mining_height, err);
                    continue
                }
            }
```

**Review by lpgeiger — CHANGES_REQUESTED** [review]
Overall looks good, just please check if there is code duplication from the original backlog sending. The backlog always gets sent at the beginning of the miner app starting. So it should just use the same method.

File: ol/miner/src/block.rs (hunk 1 above)
Comment (VenkatTeja) [reply ol/miner/src/block.rs:137]: The backlog method might need some modifications though. Will do and push.

Comment (VenkatTeja) [reply ol/miner/src/block.rs:121]: Closed

File: ol/miner/src/commit_proof.rs (hunk 2 above)
Comment (VenkatTeja) [reply ol/miner/src/commit_proof.rs]: Closed

File: ol/miner/src/block.rs (hunk 3 above)
Comment (VenkatTeja) [reply ol/miner/src/block.rs]: Closed

**Review by lpgeiger — APPROVED** [review]

**Pull request merged**